sky/cli.py

```diff
@@ -4081,10 +4081,10 @@ def spot_cancel(name: Optional[str], job_ids: Tuple[int], all: bool, yes: bool):
       # Cancel managed spot jobs with IDs 1, 2, 3
       $ sky spot cancel 1 2 3
     """
-    _, handle = backend_utils.is_controller_up(
+    controller_status, _ = backend_utils.is_controller_up(
         controller_type=controller_utils.Controllers.SPOT_CONTROLLER,
         stopped_message='All managed spot jobs should have finished.')
-    if handle is None:
+    if controller_status in [status_lib.ClusterStatus.STOPPED, None]:
```

Comment:
Add a comment: Allow INIT to proceed. 

(same below?)

What if there's some genuine problems? E.g., ssh into an UP controller, kill Ray, local status refresh to make ctrl go into INIT, then call sky cancel?

Reply:
Added the comment here.

Will test the those cases more : )

Reply:
When the controller is actually abnormal, we will still try to ssh into the controller and try to run the same command and retrieve the results.